Drivers/Comms/Src/Xbee.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Xbee.cpp
+ *
+ *  Created on: Aug 20, 2022
+ *      Author: Aaditya
+ */
+
+#include "Xbee.hpp"
+#include "main.h"
+#include "usart.h"
+
+#define UART_TIMEOUT 100
+
+#define DRONE_BASIC_PAYLOAD_TYPE 1
+
+void changeEndian(uint8_t* val, int size) {
+	uint8_t temp;
+	for (int x = 0; x < size/2; x++) {
+		temp = val[x];
+		val[x] = val[size - 1 - x];
+		val[size - 1 - x] = temp;
+	}
+}
+
+void TransmitMessage(DroneBasicPayload* payload) {
+	Packet packet = {};
+	TransmitFrameHeader tf = {};
+
+	packet.time = HAL_GetTick();
+	packet.payloadType = DRONE_BASIC_PAYLOAD_TYPE;
+
+	uint8_t checksum = 0xFF;
+
+	uint8_t rawPacket[sizeof(TransmitFrameHeader) + sizeof(Packet) + sizeof(DroneBasicPayload) + sizeof(checksum)];
+
+
+	uint8_t* tfArr = (uint8_t*)&tf;
+	uint8_t* packetArr = (uint8_t*)&packet;
+	uint8_t* payloadArr = (uint8_t*)payload;
```

Comment:
Just so I know what's going on, you are using the struct to access specific information but then you are using the pointer (of the struct) as an array of bytes so you can nicely pack it into the `rawPacket` of information for transmission?

Reply:
yeah, they were structs as first, because its easier to define and easier to read and write, but theyre then converted to an array of bytes so that we can send the packet over UART to the xbee.